app/services/promotional_tweet/post_promotional_tweet.rb

```diff
@@ -37,11 +37,11 @@ def post_tweet
 
   def post_tweet_with_image
     temp_file = create_temp_file
+    temp_file.close
     file = File.open(temp_file.path)
     client.update_with_media(message, file)
   ensure
     file.close
-    temp_file.close
```

Comment:
Can you check that after this processes that the image doesn't stay in the `/tmp` folder? Otherwise, the following will probably do the trick:

```ruby
temp_image.file.close
temp_image.file.unlink
```